CoffeeMachine_Task1/src/bg/coffee_machine/BasicCoffeeMachine.java

```diff
@@ -0,0 +1,38 @@
+package bg.coffee_machine;
+
+public class BasicCoffeeMachine implements CoffeeMachine{
+    private BasicContainer container;
+
+
+    public BasicCoffeeMachine() {
+        this.container = new BasicContainer();
+    }
+
+    @Override
+    public Product brew(Beverage beverage) {
+        if(beverage instanceof Espresso)
```

Comment:
Consider using `Beverage`'s methods for how much it costs to make the product. Currently hard-coded values are used in several different places. Also try to remove `enough*`  methods for checking supplies.